codecov_cli/commands/labelanalysis.py

```diff
@@ -313,18 +324,50 @@ def _dry_run_output(
     runner: LabelAnalysisRunnerInterface,
     dry_run_output_path: Optional[pathlib.Path],
 ):
-    labels_to_run = list(
-        set(
-            result.absent_labels
-            + result.global_level_labels
-            + result.present_diff_labels
-        )
+    labels_to_run = set(
+        result.absent_labels + result.global_level_labels + result.present_diff_labels
+    )
+    labels_skipped = set(result.present_report_labels) - labels_to_run
+    # If the test label can contain spaces and dashes the test runner might
+    # interpret it as an option and not a label
+    # So we wrap it in doublequotes just to be extra sure
+    labels_run_wrapped_double_quotes = sorted(
+        map(lambda l: '"' + l + '"', labels_to_run)
+    )
+    labels_skip_wrapped_double_quotes = sorted(
+        map(lambda l: '"' + l + '"', labels_skipped)
+    )
+
+    output_as_dict = dict(
+        runner_options=runner.dry_run_runner_options,
+        ats_tests_to_run=labels_run_wrapped_double_quotes,
+        ats_tests_to_skip=labels_skip_wrapped_double_quotes,
     )
-    output = runner.dry_run_runner_options + sorted(labels_to_run)
     if dry_run_output_path is not None:
         with open(dry_run_output_path, "w") as fd:
-            fd.write(" ".join(output) + "\n")
-    click.echo(f"ATS_TESTS_TO_RUN=\"{' '.join(output)}\"")
+            fd.write(
+                " ".join(
+                    runner.dry_run_runner_options + labels_run_wrapped_double_quotes
+                )
+                + "\n"
+            )
+        with open(str(dry_run_output_path) + "_skipped", "w") as fd:
+            fd.write(
+                " ".join(
+                    runner.dry_run_runner_options + labels_skip_wrapped_double_quotes
+                )
+                + "\n"
+            )
+        with open(str(dry_run_output_path) + ".json", "w") as fd:
+            fd.write(json.dumps(output_as_dict) + "\n")
+
+    click.echo(json.dumps(output_as_dict))
+    click.echo(
+        f"ATS_TESTS_TO_RUN={' '.join(runner.dry_run_runner_options + labels_run_wrapped_double_quotes)}"
+    )
+    click.echo(
+        f"ATS_TESTS_TO_SKIP={' '.join(runner.dry_run_runner_options + labels_skip_wrapped_double_quotes)}"
+    )
```

Comment:
This stdout is nearly but not quite json parseable.
Most tools will do one of two things:
 
  1. print json on stdout, and non-json on stderr
  2. print a human-friendly form by default and only json if given --json
  
  If you choose one of these, the `--dry-run-output-path` option can go away: `tee(1)` fills that need, then.